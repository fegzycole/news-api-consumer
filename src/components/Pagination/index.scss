@import "../../styles/mixins";

.Pagination_Btn {
  @include standard-margin-bottom(15px);
  @include transform-uppercase;
  @include text-styles(700, 11px, $secondary-color, 18px, 0.15em);
  border: 1px solid $secondary-color;
  border-radius: 18px;
  padding: 8px 32px;
}

.Pagination_Section {
  text-align: center;
}
