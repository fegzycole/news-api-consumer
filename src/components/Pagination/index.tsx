import React from "react";

import './index.scss';

export interface PaginationBtnProps {
  handleClick: () => void;
}

const PaginationBtn: React.FC<PaginationBtnProps> = ({ handleClick }) => {
  return (
    <section className="Pagination_Section">
      <button className="Pagination_Btn" onClick={handleClick}>
        Load More
      </button>
    </section>
  );
};

export default PaginationBtn;
