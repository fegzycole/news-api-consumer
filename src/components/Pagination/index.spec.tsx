import { render } from "@testing-library/react";

import Pagination, { PaginationBtnProps } from "./index";

const mockProps: PaginationBtnProps = {
  handleClick: jest.fn(),
};

describe("<Pagination />", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should match the Pagination component snapshot", () => {
    const view = render(<Pagination {...mockProps} />);
    expect(view.container).toMatchSnapshot();
  });
});
