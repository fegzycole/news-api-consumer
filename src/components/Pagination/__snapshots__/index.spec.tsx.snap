// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should match the Pagination component snapshot 1`] = `
<div>
  <section
    class="Pagination_Section"
  >
    <button
      class="Pagination_Btn"
    >
      Load More
    </button>
  </section>
</div>
`;
